src/games/strategy/triplea/ui/MapRouteDrawer.java

```diff
@@ -6,295 +6,192 @@
-              (int) (points[numTerritories - 1].y + textyOffset - yOffset - translateY), null);
-        }
-        final Image cursorImage = routeDescription.getCursorImage();
-        if (cursorImage != null) {
-          graphics.drawImage(cursorImage, (int) (points[numTerritories - 1].x + cursorXOffset - xOffset),
-              (int) (points[numTerritories - 1].y + cursorYOffset - yOffset), null);
-          if (scrollWrapX /* && !scrollWrapY */) {
-            graphics.drawImage(cursorImage, (int) (points[numTerritories - 1].x + cursorXOffset - xOffset + translateX),
-                (int) (points[numTerritories - 1].y + cursorYOffset - yOffset), null);
-            graphics.drawImage(cursorImage, (int) (points[numTerritories - 1].x + cursorXOffset - xOffset - translateX),
-                (int) (points[numTerritories - 1].y + cursorYOffset - yOffset), null);
-          }
-          if (/* !scrollWrapX && */scrollWrapY) {
-            graphics.drawImage(cursorImage, (int) (points[numTerritories - 1].x + cursorXOffset - xOffset),
-                (int) (points[numTerritories - 1].y + cursorYOffset - yOffset + translateY), null);
-            graphics.drawImage(cursorImage, (int) (points[numTerritories - 1].x + cursorXOffset - xOffset),
-                (int) (points[numTerritories - 1].y + cursorYOffset - yOffset - translateY), null);
-          }
-          if (scrollWrapX && scrollWrapY) {
-            graphics.drawImage(cursorImage, (int) (points[numTerritories - 1].x + cursorXOffset - xOffset + translateX),
-                (int) (points[numTerritories - 1].y + cursorXOffset - yOffset + translateY), null);
-            graphics.drawImage(cursorImage, (int) (points[numTerritories - 1].x + cursorXOffset - xOffset - translateX),
-                (int) (points[numTerritories - 1].y + cursorXOffset - yOffset - translateY), null);
-          }
-        }
-      }
-    } finally {
-      graphics.setTransform(original);
+    // set thickness and color of the future drawings
+    graphics.setStroke(new BasicStroke(3.5f, BasicStroke.CAP_ROUND, BasicStroke.JOIN_ROUND));
+    graphics.setPaint(Color.red);
+    graphics.setRenderingHint(RenderingHints.KEY_ANTIALIASING, RenderingHints.VALUE_ANTIALIAS_ON);
+    if (routeDescription == null) {
+      return;
+    }
+    final Route route = routeDescription.getRoute();
+    if (route == null) {
+      return;
+    }
+    final int numTerritories = route.getAllTerritories().size();
+    final Point[] points = getRoutePoints(routeDescription, mapData);
+    final boolean tooFewTerritories = numTerritories <= 1;
+    final boolean tooFewPoints = points.length <= 2;
+    final int xOffset = view.getXOffset();
+    final int yOffset = view.getYOffset();
+    final double scale = view.getScale();
+    final int jointsize = 10;
+    if (tooFewTerritories || tooFewPoints) {
+      drawLineWithTranslate(graphics, new Line2D.Float(routeDescription.getStart(), routeDescription.getEnd()), xOffset,
```

Comment:
Looking at this for a second longer, I recommend collapsing the lines 54-56 together into the same helper method with a single parameter `routeDescription`, similar for the else part of thise `if/else`, but with slightly different params